File: src/depth_first_search/DFS_adjmat..cs
using System;
					
public class Program{
	public class DFS {
		bool[] visited;
		int vertex;
		int[,] mat;
		
		public DFS(int v) {
			vertex = v;
			mat = new int[vertex,vertex];
			visited = new bool[vertex];
		}
		
		public void addEdge(int u, int v){
			mat[u,v]=1;
		}
		
		public void dfs(int v){
			Console.WriteLine(v);
			visited[v]=true;
			for(int i=0;i<vertex;i++){
				if(mat[v,i]==1 && !visited[i]){
					dfs(i);
				}
			}
		}

	}
	public static void Main()
	{
		int vertex = 4;
		DFS d = new DFS(vertex);
		d.addEdge(0, 1);
		d.addEdge(0, 2);
		d.addEdge(2, 0);
		d.addEdge(1, 2);
		d.addEdge(2, 3);
		d.addEdge(3, 3);
		d.dfs(2);
	}
}

File: src/depth_first_search/DFS_adjmat.js
let mat = [[],[]];
let visited = [];
function addEdge(u, v) {
    mat[[u],[v]] = 1;
}

function dfs(s) {
    console.log(s);
    visited[s] = 1;
    for (let i = 0; i < vertex; i++) {
        if (mat[[s],[i]] == 1 && visited[i] == 0) {
            dfs(i);
        }
    }
}

let vertex = 5;
addEdge(0, 1);
addEdge(0, 2);
addEdge(1, 3);
addEdge(1, 4);
console.log(mat[[0],[1]]);

function initVisited() {
    for (let index = 0; index < vertex; index++) {
        visited[index] = 0;
    }
}
initVisited();
dfs(0);